syntax = "proto3";

option go_package = "protobuf";
package inkah;

service Inkah {
    rpc RegisterEvent (Event) returns (EventResponse) {}
}

enum EventType {
    SPAN_BEGIN = 0;
    SPAN_END = 1;
    REQUEST_BEGIN = 2;
    REQUEST_END = 3;
}

message Event {
    string traceId = 1;
    string spanId = 2;
    string parentSpanId = 3;
    string requestId = 4;

    EventType eventType = 5;
}

message EventResponse {}
